packages/framework/src/Model/Order/PromoCode/CurrentPromoCodeFacade.php

```diff
@@ -114,54 +118,55 @@ protected function validatePromoCodeByProductsInCart(PromoCode $promoCode, Cart
         if ($isValidPromoCode === false) {
             throw new PromoCodeWithoutRelationWithAnyProductFromCurrentCartException($promoCode);
         }
+
+        return $allowedProductIds;
     }
 
     /**
      * @param \Shopsys\FrameworkBundle\Model\Order\PromoCode\PromoCode $promoCode
-     * @param \Shopsys\FrameworkBundle\Model\Cart\Cart $cart
+     * @param \Shopsys\FrameworkBundle\Model\Pricing\Price $price
      */
-    protected function validateLimit(PromoCode $promoCode, Cart $cart): void
+    protected function validateLimit(PromoCode $promoCode, Price $price): void
```

Comment:
I think `$(cart|products)totalPrice` or something like that would be more descriptive parameter name (the same in getHighestLimitByPromoCodeAndTotalPrice())

Reply:
Is it bound specifically to the total Cart/Products/Total Price, or the limit validation will work for any price I want?
I would probably rather rename `getHighestLimitByPromoCodeAndTotalPrice ` to `getHighestLimitByPromoCodeAndPrice`... let's discuss please